consensus/consensus-types/src/vote.rs

```diff
@@ -108,7 +109,7 @@ impl Vote {
     }
 
     /// Return the signature of the vote
-    pub fn signature(&self) -> &bls12381::Signature {
+    pub fn signature(&self) -> &SignatureWithStatus {
```

Comment:
this probably should return the actual signature instead of the status, we can have a new function to return the status `is_verified` or something similar

Reply:
In a few places in the code, I'm actually doing something like `vote.signature().set_verified()` to set the status to verified. So, I needed the the full signaturestatus.

Reply:
Right now, in the actual code, I'm only doing it inside the verify method. I also had to set the verified flag in a lot of unit tests for which I needed this method.

Reply:
test should have its own functions for clarify (marked with #[cfg(test)])

Reply:
especially the status has interior mutability, it's very confusing to people who read the code who thinks it's immutable

Reply:
Sure. I can have a separate function to cater for unit tests.

Reply:
Done.